Sources/BraintreeCoreSwift/BTAPIHTTP.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+// TODO: Make internal once rest of core is in Swift
+@objcMembers public class BTAPIHTTP: BTHTTP {
+
+    let accessToken: String?
+
+    @objc(initWithBaseURL:accessToken:)
+    public init(url: URL, accessToken: String? = "") {
```

Comment:
Why default to an empty string? Can we make this a required param instead?

Reply:
There was some wild stuff going on where the tests where crashing, and it all hinged on the [config fetch code](https://github.com/braintree/braintree_ios/blob/91cfc9567622099af7764b512f1104af9c15a9a5/Sources/BraintreeCore/BTAPIClient.m#L303). We got crashes in both unit and integration tests when the different API and HTTP clients are instantiated without access tokens. Swift was less forgiving than Obj-C in this case.

This is a workaround – we tried some other stuff but it was turning into a rabbit hole. We felt that `BTAPIClient` constructing dependencies in `fetchOrReturnRemoteConfiguration` was a potential root cause to consider refactoring since the method does more than its name implies.

Reply:
We can make this non-optional now that we are handling this with the updated check in `fetchOrReturnRemoteConfiguration`. Good call.